### PR TITLE
cod3.d: convert to reg_t

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

File: src/dmd/backend/cod1.d [hunk 1]
```diff
@@ -3315,7 +3315,7 @@ void cdfunc(ref CodeBuilder cdb, elem* e, regm_t* pretregs)
                     regm_t tosave = keepmsk & retregs;
 
                     // tosave is the mask to save and restore
-                    for (int j = 0; tosave; j++)
+                    for (reg_t j = 0; tosave; j++)
```
Comment (thewilsonator) [inline src/dmd/backend/cod1.d]: ???

File: src/dmd/backend/cod2.d [hunk 2]
```diff
@@ -4536,12 +4536,12 @@ void cdneg(ref CodeBuilder cdb,elem *e,regm_t *pretregs)
...
         if (I32)
         {
-            uint reg = (sz == 8) ? findregmsw(retregs) : findreg(retregs);
+            const reg = (sz == 8) ? findregmsw(retregs) : findreg(retregs);
             cdb.genc2(0x81,modregrm(3,6,reg),0x80000000); // XOR EDX,sign bit
         }
         else
         {
-            uint reg = (sz == 8) ? cast(uint) AX : findregmsw(retregs);
+            const reg = (sz == 8) ? cast(uint) AX : findregmsw(retregs);
```
Comment (thewilsonator) [inline src/dmd/backend/cod2.d]: Is the cast needed anymore?

File: src/dmd/backend/cod2.d [hunk 3]
```diff
@@ -4609,12 +4609,12 @@ void cdabs(ref CodeBuilder cdb,elem *e, regm_t *pretregs)
...
         if (I32)
         {
-            int reg = (sz == 8) ? findregmsw(retregs) : findreg(retregs);
+            const reg = (sz == 8) ? findregmsw(retregs) : findreg(retregs);
             cdb.genc2(0x81,modregrm(3,4,reg),0x7FFFFFFF); // AND EDX,~sign bit
         }
         else
         {
-            int reg = (sz == 8) ? cast(int) AX : findregmsw(retregs);
+            const reg = (sz == 8) ? cast(int) AX : findregmsw(retregs);
```
Comment (thewilsonator) [inline src/dmd/backend/cod2.d]: Ditto

**Review by thewilsonator — APPROVED** [review]
Otherwise LGTM.

File: src/dmd/backend/cod1.d (hunk 1 above)
Comment (WalterBright) [reply src/dmd/backend/cod1.d]: The subsequent code uses `j` as a register number.

Comment (thewilsonator) [reply src/dmd/backend/cod1.d]: Fair enough.

**Pull request merged**